src/SFML/Graphics/RenderTarget.cpp

```diff
@@ -211,10 +211,9 @@ Vector2f RenderTarget::mapPixelToCoords(const Vector2i& point) const
 Vector2f RenderTarget::mapPixelToCoords(const Vector2i& point, const View& view) const
 {
     // First, convert from viewport coordinates to homogeneous coordinates
-    Vector2f        normalized;
     const FloatRect viewport = FloatRect(getViewport(view));
-    normalized.x             = -1.f + 2.f * (static_cast<float>(point.x) - viewport.left) / viewport.width;
-    normalized.y             = 1.f - 2.f * (static_cast<float>(point.y) - viewport.top) / viewport.height;
+    const Vector2f  normalized(-1.f + 2.f * (static_cast<float>(point.x) - viewport.left) / viewport.width,
+                              1.f - 2.f * (static_cast<float>(point.y) - viewport.top) / viewport.height);
```

Comment:
Wanted to refactor this to _not_ use individual components, but the different signs for X and Y made it a bit more complex 😕 so not sure if it's truly more readable:
```cpp
Vector2f RenderTarget::mapPixelToCoords(const Vector2i& pixel, const View& view) const {
    // These could also be inlined, at the cost of more complex expression
    const Vector2f viewPos = viewPort.getPosition();
    const Vector2f viewSize = viewPort.getSize();

    const Vector2f point(pixel);
    const Vector2f normalized = Vector2f(-1.f, 1.f) 
        + Vector2f(2.f, -2.f).cwiseMul(point - viewPos).cwiseDiv(viewSize);
    ...
```
If we had proper operators, this would be much less ugly:
```cpp
    const Vector2f normalized = Vector2f(-1.f, 1.f) + Vector2f(2.f, -2.f) * (point - viewPos) / viewSize;
```

Reply:
> Wanted to refactor this to not use individual components, but the different signs for X and Y made it a bit more complex 😕 so not sure if it's truly more readable:

I think this is an improvement. I have an easier time reasoning about these operations in a 2D space. I'd be happy to add this to the PR. Plus it means getting rid of some `static_cast`s :)

Reply:
I assume this is still open?

Reply:
I just added that to the PR.